zutil.c

```diff
@@ -154,6 +154,14 @@ void Z_INTERNAL PREFIX3(free_aligned)(zng_cfree_func zfree, void *opaque, void *
     void *original_ptr = (void *)((uintptr_t)ptr - sizeof(void *));
     void *free_ptr = *(void **)original_ptr;
 
+    /* Validate original_ptr, the distance to ptr should be less than double the maximum alignment of 64 bytes */
+    ptrdiff_t dist = original_ptr - free_ptr;
+    if (dist < sizeof(void *) || dist > 127) {
+       Tracev((stderr, "free_aligned: Allocation/deallocation mismatch"));
+       zfree(opaque, ptr);
```

Comment:
Should this be `PREFIX(zcfree)(opaque, ptr);` as above?

Reply:
It should still use the custom free function if it is passed, otherwise it might fail `alloc_64` test...

The distance with `zalloc_64` test is 7, so we can't limit the low end to `sizeof(void *)`.